OnDisconnect not call

https://github.com/eclipse/paho.golang/blob/76a1f9a081d13c3e6da7706c7dd3e24c8139d82c/paho/client.go#L378-L387

Only receive `packets.DISCONNECT` call `OnDisconnect()`

Network error not call `OnDisconnect()`

```
[DEBUG]: 2021/01/19 12:15:41 client.go:622: sending QoS0 message
[DEBUG]: 2021/01/19 12:15:43 client.go:400: error called: EOF
[DEBUG]: 2021/01/19 12:15:43 client.go:408: client stopped
[DEBUG]: 2021/01/19 12:15:43 client.go:410: ping stopped
[DEBUG]: 2021/01/19 12:15:43 client.go:412: conn closed
```

But Network error cannot be known

**Need notified to Network error**

--- a/paho/client.go
+++ b/paho/client.go
@@ -34,7 +34,10 @@ type (
 		Persistence   Persistence
 		PacketTimeout time.Duration
 		OnDisconnect  func(*Disconnect)
-		PublishHook   func(*Publish)
+		// Only receive packets.DISCONNECT call
+		OnClientError func(error)
+		// Client error call, For example: net.Error
+		PublishHook func(*Publish)
 		// PublishHook allows a user provided function to be called before
 		// a Publish packet is sent allowing it to inspect or modify the
 		// Publish, an example of the utility of this is provided in the
@@ -126,6 +129,9 @@ func NewClient(conf ClientConfig) *Client {
 		c.PingHandler = DefaultPingerWithCustomFailHandler(func(e error) {
 			c.Error(e)
 		})
+	}
+	if c.OnClientError == nil {
+		c.OnClientError = func(e error) {}
 	}
 
 	return c
@@ -410,6 +416,7 @@ func (c *Client) Error(e error) {
 	c.debug.Println("ping stopped")
 	c.Conn.Close()
 	c.debug.Println("conn closed")
+	c.OnClientError(e)
 	c.mu.Unlock()
 }
 

